Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,330 @@
+                "Awaiting remaining {} inflight requests".format(self._inflight_counter)
+            )
+            await asyncio.sleep(5)
+
+        self.logger.log_info("Shutdown complete")
+
+    def make_sample_params(self, request: CompletionRequest) -> SamplingParams:
+        """
+        Creates a vLLM SamplingParams object from the given arguments.
+        """
+        return SamplingParams(
+            n=1,
+            best_of=1,
+            temperature=request.temperature,
+            top_p=request.top_p,
+            top_k=request.top_k,
+            stop=request.stop,
+            ignore_eos=request.ignore_eos,
+            max_tokens=request.max_tokens,
+        )
+
+    def make_output_options(self, req: CompletionRequest) -> OutputOptions:
+        """
+        Creates an OutputOptions object from the given arguments.
+        """
+        return OutputOptions(
+            echo=req.echo,
+            stream=req.stream,
+        )
+
+    def preprocess(self, request) -> Tuple[str, SamplingParams, OutputOptions]:
+        """
+        Converts a Triton request into a VLLM input by extracting the prompt and sampling parameters.
+        """
+        # deserialize the VLLM request from the Triton InferRequest
+        try:
+            serialized_json = pb_utils.get_input_tensor_by_name(
+                request, "serialized_request_json"
+            ).as_numpy()[0]
+            request = CompletionRequest(**json.loads(serialized_json))
+
+            return (
+                request.prompt,
+                self.make_sample_params(request),
+                self.make_output_options(request),
+            )
+        except Exception as e:
+            self.logger.log_info(f"Error parsing request: {e}")
+            raise e
+
+    def postprocess(
+        self, output: VLLMOutput, output_options: OutputOptions, inflight_avg: float
+    ):
+        """
+        Converts a VLLM output into a Triton response.
+        """
+        # create a Triton Response from the vLLM output
+        response = CompletionResponse()
+        response.request_id = output.request_id
+        response.finished = output.finished
+
+        # number of tokens in the prompt and total number of tokens generated across all completions
+        response.prompt_token_count = len(output.prompt_token_ids)
+
+        # optionally echo the prompt back to the client
+        if output_options.echo:
+            response.prompt = output.prompt
+
+        for completion in output.outputs:
+            completion_output = Completion(
+                index=completion.index,
+                text=completion.text,
+                finish_reason=completion.finish_reason,
+                cumulative_logprob=completion.cumulative_logprob,
+                gen_token_count=len(completion.token_ids),
+            )
+            response.completions.append(completion_output)
+
+        response.current_inflight_count = self._inflight_counter
```

Comment:
see below about the possibility of counting pending requests here as inflight, option is to use:
```
 len(self.llm.engine.scheduler.running)
```